react-ui/src/ControlPanel.jsx

```diff
@@ -55,13 +60,28 @@ export class ControlPanel extends PureComponent {
     this.setState({ open: !this.state.open })
   }
 
-  setPictureFilter = ({ value }) => {
+  setPictureFilter = (value) => {
     this.props.setPictureFilter(value)
     this.props.captureUserPicture()
   }
+
+  setPictureFilterWeight = ({ target }) => {
+    this.props.setPictureFilterWeight(target.value)
+    this.props.captureUserPicture()
+  }
+
+  newPictureFilterFromImage = ({ target }) => {
```

Comment:
Usually methods like this are named `handleX` or `setX` - to better distinguish between render functions, functions which calculate something and something which then handles user input.

In this instance `handleNewFilterImage` would make a good name